### PR TITLE
tests/coalaTest.py: Add test case for ignore flag

### PR DESCRIPTION
**Pull request by Vedarth**
Adds a temporary file, then immediately ignores it.
Ideally, coala should ignore the file and do nothing.

Closes https://github.com/coala/coala/issues/4886

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: tests/coalaTest.py [hunk 1]
```diff
@@ -322,3 +322,18 @@ def test_coala_without_color(self):
...
+                prepare_file(['#fixme'], None) as (lines, filename):
+            retval, stdout, stderr = execute_coala(
+                    coala.main, 'coala',
+                    '-c', os.devnull,
+                    '--non-interactive',
+                    '-f', filename,
+                    '-i', filename,
+                    '--ignore', filename,
+                    '-b', 'LineCountTestBear')
+            self.assertIn(stdout, 'Executing section cli...\n')
```
Comment (nityeshaga) [inline tests/coalaTest.py]: Are you sure this line would work? 
In [#4930 (review)](https://github.com/coala/coala/pull/4930#pullrequestreview-90337947), @userzimmermann said that the code should explicitly check that **nothing more** than 

> Executing section cli...

appears in stdout whereas I think your code would just check if that line is present in the output.

Comment (Vedarth) [reply tests/coalaTest.py]: You are right. I fixed it now.

Comment (nityeshaga) [reply tests/coalaTest.py]: Great! Now we just need to figure the `LineCountTestBear` thingy.

Comment (nityeshaga) [reply tests/coalaTest.py]: Also somehow there have been 5000+ additions to the code. I am not sure why.

Comment (Vedarth) [reply tests/coalaTest.py]: I think I know. I will fix it as soon as I get to my computer. On my phone atm.

**Comment by Vedarth** [comment]
Actually on further pondering over the issue I realized that my test will run irrespective of the bear used as it is going to display `Executing section cli...\n` if `-i` or `--ignore` is used. So I think it should be fine using a bear to check working of ignore flag as far as we are only concentrating on the output after use of ignore flag.
What are your opinions @userzimmermann ?

File: tests/coalaTest.py [hunk 2]
```diff
@@ -322,3 +322,18 @@ def test_coala_without_color(self):
...
+
+    def test_coala_ignore_file(self):
+        with bear_test_module(), \
+                prepare_file(['#fixme'], None) as (lines, filename):
+            retval, stdout, stderr = execute_coala(
+                    coala.main, 'coala',
+                    '-c', os.devnull,
+                    '--non-interactive',
+                    '-f', filename,
+                    '-i', filename,
```
Comment (jayvdb) [inline tests/coalaTest.py]: why `-i` and `--ignore` ?

Comment (Vedarth) [reply tests/coalaTest.py]: To test the ignore flag as defined in https://github.com/coala/coala/issues/4886#issue-275183478

Comment (jayvdb) [reply tests/coalaTest.py]: they are two flags which have the same impact, and it is bizarre to use both together.  Nobody would do this, and a test doing it proves nothing.

Comment (Vedarth) [reply tests/coalaTest.py]: Alright. I will write a different test for `--ignore` :sweat_smile:

Comment (jayvdb) [reply tests/coalaTest.py]: What part of coala are you testing better by having a test for `-i` and `--ignore`.
Which lines of code?

Comment (Vedarth) [reply tests/coalaTest.py]: This test checks the working of ignore flag which can be used as `-i` or `--ignore`, which is a cmdline option. So we need to test both of them.

The definition of this  cmdline option is stored in https://github.com/coala/coala/blob/4e0c88e5d5e575e6fd20b5abb437d0ef54cb6077/coalib/parsing/DefaultArgParser.py#L148-L150

Comment (Vedarth) [reply tests/coalaTest.py]: You are right. Irrespective of the `add_argument` same code will get executed.

Comment (Vedarth) [reply tests/coalaTest.py]: I think a single test should suffice the task.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5138/commits/e74828570d2e2f0e12dcfdf3e1574e2058097a1b

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**